net/icmp/icmp_ioctl.c

```diff
@@ -0,0 +1,90 @@
+/****************************************************************************
+ * net/icmp/icmp_ioctl.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <debug.h>
+#include <errno.h>
+
+#include <net/if.h>
+
+#include <nuttx/fs/ioctl.h>
+#include <nuttx/mm/iob.h>
+#include <nuttx/net/net.h>
+
+#include "icmp/icmp.h"
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: icmp_ioctl
+ *
+ * Description:
+ *   This function performs icmp specific ioctl() operations.
+ *
+ * Parameters:
+ *   conn     The ICMP connection of interest
+ *   cmd      The ioctl command
+ *   arg      The argument of the ioctl cmd
+ *   arglen   The length of 'arg'
+ *
+ ****************************************************************************/
+
+int icmp_ioctl(FAR struct socket *psock,
+               int cmd, FAR void *arg, size_t arglen)
+{
+  FAR struct icmp_conn_s *conn = psock->s_conn;
+  int ret = OK;
+
+  net_lock();
+
+  switch (cmd)
+    {
+      case FIONREAD:
+        if (iob_peek_queue(&conn->readahead) != NULL)
+          {
+            *(FAR int *)((uintptr_t)arg) =
```

Comment:
compiler may generate the warning if we directly cast unsigned long to the pointer on some platform.

Reply:
Just follow the existing code format...

Reply:
@xiaoxiang781216 it is cast from `FAR void *`, but since similar is present in other places and does not bring any harm let's go with it. Maybe I will wipe it out in one of the next PRs